packages/base/src/commands.ts

```diff
@@ -285,6 +290,190 @@ export function addCommands(
+      };
+
+      const model = tracker.currentWidget?.model;
+      const localState = model?.sharedModel.awareness.getLocalState();
+      if (!model || !localState || !localState['selected']?.value) {
+        return;
+      }
+      const localSelectedLayer = localState['selected'].value;
+      const localSelectedLayerId = Object.keys(localSelectedLayer)[0];
+
+      // Open form and get user input
+      const formValues = await new Promise<IDict>(resolve => {
+        const dialog = new ProcessingFormDialog({
+          title: 'Buffer',
+          schema: schema,
+          model: tracker.currentWidget?.model as IJupyterGISModel,
+          sourceData: {
+            inputLayer: localSelectedLayerId,
+            bufferDistance: 10,
+            projection: 'EPSG:4326'
+          },
+          cancelButton: false,
+          syncData: (props: IDict) => {
+            resolve(props);
+            dialog.dispose();
+          }
+        });
+
+        dialog.launch();
+      });
+
+      if (!formValues) {
+        return;
+      }
+
+      const bufferDistance = formValues.bufferDistance;
+      const selectedLayerId = formValues.inputLayer;
+      const selectedLayer = layers[selectedLayerId];
+
+      if (!selectedLayer.parameters) {
+        console.error('Selected layer not found.');
+        return;
+      }
+
+      const sourceId = selectedLayer.parameters.source;
+      const source = sources[sourceId];
+
+      if (!source.parameters) {
+        console.error(`Source with ID ${sourceId} not found or missing path.`);
+        return;
+      }
+
+      let geojsonString: string;
+
+      if (source.parameters.path) {
+        const fileContent = await loadFile({
+          filepath: source.parameters.path,
+          type: source.type,
+          model: tracker.currentWidget?.model as IJupyterGISModel
+        });
+
+        geojsonString =
+          typeof fileContent === 'object'
+            ? JSON.stringify(fileContent)
+            : fileContent;
+      } else if (source.parameters.data) {
+        geojsonString = JSON.stringify(source.parameters.data);
+      } else {
+        throw new Error(
+          `Source ${sourceId} is missing both 'path' and 'data' parameters.`
+        );
+      }
+
+      const fileBlob = new Blob([geojsonString], {
+        type: 'application/geo+json'
+      });
+      const geoFile = new File([fileBlob], 'data.geojson', {
+        type: 'application/geo+json'
+      });
+
+      const Gdal = await getGdal();
+      const result = await Gdal.open(geoFile);
+
+      if (result.datasets.length > 0) {
+        const dataset = result.datasets[0] as any;
+        const layerName = dataset.info.layers[0].name;
+
+        const sqlQuery = `
+          SELECT ST_Union(ST_Buffer(geometry, ${bufferDistance})) AS geometry, *
+          FROM "${layerName}"
+        `;
+
+        const options = [
+          '-f',
+          'GeoJSON',
+          '-t_srs',
+          'EPSG:4326',
```

Comment:
Use the projection the user set from the UI

```suggestion
          formValues.projection,
```